internal/npm_install/generate_build_file.js

```diff
@@ -660,8 +647,9 @@ function filterFilesForFilegroup(files, exts = []) {
       // include files with no extensions if noExt is true
       if (allowNoExts && !path.extname(f)) return true;
       // filter files in exts
+      const lc = f.toLowerCase();
       for (const e of exts) {
-        if (e && f.endsWith(e)) {
+        if (e && lc.endsWith(e.toLowerCase())) {
```

Comment:
May as well make this case insensitive as there are probably going to be npm packages with`.JS` files out there somewhere